sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1961,6 +1961,14 @@ object SQLConf {
     .intConf
     .createWithDefault(25)
 
+  val MAX_DATA_SOURCE_SCAN_METADATA_STRING_LENGTH =
+    buildConf("spark.sql.maxDataSourceScanMetadataStringLength")
+      .doc("Maximum string length for one metadata of `DataSourceScanExec`. This is used in " +
```

Comment:
Is this only for `DataSourceScanExec`? seems the scope is too narrow. Can we find some other similar instances?

Reply:
yes, currently it is only used in DataSourceScanExec. I checked the implementations of TreeNode.simpleString method, now, no other instance has metadata contained in simpleString.
I just refined the configuration name and description to make it be able to cover bigger scope if needed.